rct229/rulesets/ashrae9012019/ruleset_functions/get_hvac_sys_and_assoc_zones_largest_exhaust_source.py

```diff
@@ -0,0 +1,91 @@
+    """
+    Returns a list with the sum of the hvac fan system exhaust fan cfm values, the maximum zone level exhaust source
+    across the zones associated with the HVAC system, the number of exhaust fans associated with the hvac fan system,
+    and the maximum cfm of all of the exhaust fans associated with the hvac system fan system [
+    hvac_sys_exhaust_cfm_sum, maximum_zone_exhaust, num_hvac_exhaust_fans, maximum_hvac_exhaust]. This is for
+    evaluating G3.1.2.10 exception 6 which states "Where the largest exhaust source is less than 75% of the design
+    outdoor airflow. This exception shall only be used if exhaust air energy recovery is not used in the proposed
+    design.
+
+    Parameters
+    ----------
+    rmi dict
+    Dictionary of a rule model instance object in which the largest exhaust source CFM is to be returned.
+
+    hvac_sys_id str
+    The hvac system object id for which the largest exhaust source is being determined.
+
+    Returns
+    -------
+    get_hvac_sys_and_assoc_zones_largest_exhaust_source: dict
+    A dictionary with the sum of the hvac fan system exhaust fan cfm values, the maximum zone level exhaust source
+    across the zones associated with the HVAC system, the number of exhaust fans associated with the hvac fan system,
+    and the maximum cfm of all of the exhaust fans associated with the hvac system fan system
+    {
+    hvac_fan_sys_exhaust_sum: 3000,
+    maximum_zone_exhaust: 500,
+    num_hvac_exhaust_fans: 2,
+    maximum_hvac_exhaust: 3000
+    }.
+    This is for evaluating G3.1.2.10 exception 6 which states "Where the largest exhaust source is less than 75% of the
+    design outdoor airflow. This exception shall only be used if exhaust air energy recovery is not used in the proposed
+    design.
+    """
+    zones_and_terminal_units_served_by_hvac_sys_dict = (
+        get_dict_of_zones_and_terminal_units_served_by_hvac_sys(rmi)
+    )
+    assert_(
+        hvac_sys_id in zones_and_terminal_units_served_by_hvac_sys_dict.keys(),
+        f"HVAC {hvac_sys_id} does not "
+        f"have terminals and zones "
+        f"connected with. Check inputs!",
+    )
+
+    # HVAC level exhaust fan
+    hvac = find_exactly_one_hvac_system(rmi, hvac_sys_id)
+    hvac_fan_sys_exhaust_list = find_all("$.fan_system.exhaust_fans[*]", hvac)
+    hvac_fan_sys_exhaust_flow_list = [
+        exhaust_fan.get("design_airflow", ZERO.FLOW)
+        for exhaust_fan in hvac_fan_sys_exhaust_list
+    ]
+
+    # zone level exhaust fan
+    zone_id_list = zones_and_terminal_units_served_by_hvac_sys_dict[hvac_sys_id][
+        "zone_list"
+    ]
+    zone_fan_exhaust_flow_list = list(
+        filter(
+            None,
+            [
+                find_one(
+                    "$.zonal_exhaust_fan.design_airflow",
+                    find_exactly_one_zone(rmi, zone_id),
+                )
+                for zone_id in zone_id_list
+            ],
+        )
+    )
+
+    return {
```

Comment:
I see that you removed the `cfm` from the return dict key.  That is good.  The RDS should be changed to match.